packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -62,138 +95,208 @@ function WorklistDataGrid(props) {
+        className={isRowHeader
+          ? cx(['worklist-data-grid-row-header',
+            { 'worklist-data-grid-cell-selected': isSelected }])
+          : cx(['worklist-data-grid-cell-data',
+            { 'worklist-data-grid-cell-selected': isSelected }])}
+        onCellSelect={onCellSelect}
+        onCellSelectionChange={handleCellSelectionChange}
+        onMoveCellFocus={handleMoveCellFocus}
+        onNavigationModeChange={handleNavigationModeChange}
+        isRowSelectionModeEnabled={hasSelectableRows}
+        rowsLength={rows.length}
+        columnsLength={displayedColumns.length}
+        isRowHeader={isRowHeader}
+        isNavigationEnabled={isNavigationEnabled}
+      >
+        {cell?.content}
+      </Cell>
+    );
+  };
+
+  const buildColumn = (columnData, columnIndex) => {
+    const acceptsFocus = focusedRow.current === 0 && focusedCol.current === columnIndex;
+    const width = `${columnData.width || props.columnWidth}px`;
     const height = props.columnHeaderHeight;
     return (
-      /* eslint-disable react/forbid-dom-props */
-      <th key={columnData.id} className={cx('worklist-data-grid-column-header')} tabIndex="-1" style={{ width, height }}>{columnData.displayName}</th>
+      <ColumnHeaderCell
+        columnId={`${columnData.id}`}
+        coordinates={{ x: 0, y: columnIndex }}
+        acceptsFocus={acceptsFocus}
+        width={width}
+        height={height}
+        rowsLength={rows.length}
+        columnsLength={displayedColumns.length}
+        onMoveCellFocus={handleMoveCellFocus}
+        onCellSelectionChange={handleCellSelectionChange}
+      >
+        {columnData.displayName}
+      </ColumnHeaderCell>
     );
   };
 
   const buildColumns = (allColumns) => {
     if (allColumns?.length > 0) {
       return (
         <tr height={props.columnHeaderHeight}>
-          {allColumns.map(columnData => (buildColumn(columnData)))}
+          {allColumns.map((columnData, columnIndex) => (buildColumn(columnData, columnIndex)))}
         </tr>
       );
     }
     return undefined;
   };
 
-  const buildRow = (row) => {
-    const height = row.height || props.rowHeight;
+  const buildRow = (row, rowIndex) => {
+    const columnIndexOffSet = hasSelectableRows ? 1 : 0;
+
     return (
-      <tr key={row.id} className={cx('worklist-data-grid-row')} style={{ height }}>
+      <Row
+        id={row.id}
+        height={props.rowHeight}
+        isSelected={row.isSelected}
+        onCellSelectionChange={setCurrentSelectedCell}
+        isRowSelectionModeEnabled={hasSelectableRows}
+        onRowSelect={onRowSelect}
+        onClearRowSelection={clearRowSelection}
+      >
+        {hasSelectableRows && getRowSelectionCellData(rowIndex, row)}
         {row.cells.map((cell, cellColumnIndex) => (
-          getCellData(cell, cellColumnIndex)
+          getCellData(rowIndex, cellColumnIndex + columnIndexOffSet, cell, row.id)
```

Comment:
The Row React component should be creating all of its cells.  That logic should no longer reside in the WorklistDataGrid.